### PR TITLE
Adding triton acceptable GraphSage model for generating node embeddings

### PR DESCRIPTION
**Pull request by sachinsharma9780**
I have added a GraphSage model trained on obgn-products dataset. The folder contains the following:

1.  How to trace a PyTorchGeometric model in order to generate a Triton acceptable model.
2. Inference script which interacts with GraphSage model deployed on triton inference server. In this script, we send a random test node's node feature matrix and adjacency list to triton server for the inference and receive backs its corresponding node embeddings.
3. It also includes model's config file 
4. 'model.pt': It is a traced model from the learned weights (checkpoint) generated after performing trace on GraphSage model
5. Currently it support batch size of 1

### CODE REVIEW

**Comment by geenen124** [comment]
@sachinsharma9780 it looks like the commits for the graphsage model are missing from this PR?

File: trace_model/obgn-products/trace_obgn-product_graphsage_model.ipynb [hunk 1]
```diff
@@ -0,0 +1,546 @@
...
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "torch.Size([1000, 100])\n"
+     ]
+    }
+   ],
+   "source": [
+    "# padding nodes\n",
+    "max_nodes = 1000\n",
```
Comment (geenen124) [inline trace_model/obgn-products/trace_obgn-product_graphsage_model.ipynb:274]: To flexibly calculate the maximum possible neighborhood size for padding if we want to vary the neighborhood sizes, we may want to introduce a convenience function like below. For the current model (including self loops, so `n+1` per hop) the total size is `1056` - we may want to rerun the trace for this, wdyt?

```
import math

def neighborhood_sizes_to_max_nodes(sizes_list):
    return math.prod([s+1 for s in sizes_list])
```

Comment (sachinsharma9780) [reply trace_model/obgn-products/trace_obgn-product_graphsage_model.ipynb:274]: @geenen124 need to test this scenario, will let you know!!
